base/expr.jl

```diff
@@ -109,13 +109,17 @@ evaluates expressions in that module.
 Core.eval
 
 """
-    @eval
+    @eval [m,] ex
 
-Evaluate an expression and return the value.
+Evaluate an expression with values interpolated into it using `eval`.
+If two arguments are provided, the first is the module to evaluate in.
 """
 macro eval(x)
     :($(esc(:eval))($(Expr(:quote,x))))
 end
+macro eval(m, x)
+    :($(esc(:eval))($(esc(m)),$(Expr(:quote,x))))
```

Comment:
I think this is this way because of a few things:

* Every module has its own `eval` that already evaluates into that module. I believe this can only be avoided by creating a `baremodule`, which is rare, of course.
* Is emitting the code `eval(current_module(), expr)` really equivalent to `eval(expr)`? I used to think I understood `current_module()` but you showed me some behavior that made me confused and think that I don't really understand what it does anymore.